storage/badger/chunkDataPacks.go

```diff
@@ -67,25 +69,42 @@ func (ch *ChunkDataPacks) Remove(chunkID flow.Identifier) error {
 }
 
 func (ch *ChunkDataPacks) BatchStore(c *flow.ChunkDataPack, batch storage.BatchStorage) error {
+	sc := toStoredChunkDataPack(c)
 	writeBatch := batch.GetWriter()
 	batch.OnSucceed(func() {
-		ch.byChunkIDCache.Insert(c.ChunkID, c)
+		ch.byChunkIDCache.Insert(sc.ChunkID, sc)
 	})
-	return operation.BatchInsertChunkDataPack(c)(writeBatch)
+	return operation.BatchInsertChunkDataPack(sc)(writeBatch)
 }
 
-func (ch *ChunkDataPacks) ByChunkID(chunkID flow.Identifier) (*flow.ChunkDataPack, error) {
+func (ch *ChunkDataPacks) ByChunkID(chunkID flow.Identifier) (*storagemodel.StoredChunkDataPack, error) {
```

Comment:
That's inconsistent - we `Store(c *flow.ChunkDataPack)` but retrieve `*storagemodel.StoredChunkDataPack`.
I think this should both just work on `flow.ChunkDataPack` and use `StoredChunkDataPack` internally, never leaking it outside of the package

Reply:
Agree, but the problem is in order to retrieve as `*flow.ChunkDataPack` we need to replace the `CollectionID` in the `StoredChunkDataPack` with the actual `Collection`. For that, we need to incorporate the `Collections` database into the `ChunkDataPacks` database as a member field, i.e., the `badger.ChunkDataPacks` would look like this:
```
type ChunkDataPacks struct {
	db             *badger.DB
	collections    *Collections
	byChunkIDCache *Cache
}
```

Not sure if we want to go in this direction with the extra level of dependency relations it adds, i.e., we then will both have `badger.Collections` as well as `badger.ChunkDataPacks` (that internally has `badger.Collections`) in some places. 

Just to highlight, a low-hanging fruit is to make `badger.ChunkDataPacks` both store and retrieve as `StoredChunkDataPack`, i.e., `Store(c *storagemodel.StoredChunkDataPack)`

Reply:
I see the concern, but we already do this - see https://github.com/onflow/flow-go/blob/2366de82cbc5e6c68eadbadbe246784c16fc46aa/storage/badger/blocks.go#L25 - `Blocks` depends on other data providers to serve full blocks.
As long as its all stay within the same package, I think it's OK pattern

Reply:
Fixed